services/director-v2/src/simcore_service_director_v2/api/routes/dynamic_services.py

```diff
@@ -1,20 +1,225 @@
+    )
+    if not simcore_service.needs_dynamic_sidecar:
+        # forward to director-v0
+        redirect_url_with_query = director_v0_client.client.base_url.copy_with(
+            path="/v0/running_interactive_services",
+            params={
+                "user_id": f"{service.user_id}",
+                "project_id": f"{service.project_id}",
+                "service_uuid": f"{service.node_uuid}",
+                "service_key": f"{service.key}",
+                "service_version": f"{service.version}",
+                "service_basepath": f"{service.basepath}",
+            },
+        )
+        log.debug("Redirecting %s", redirect_url_with_query)
+        return RedirectResponse(redirect_url_with_query)
+
+    # if service is already running return the status
+    if await is_dynamic_service_running(
+        dynamic_services_settings.dynamic_sidecar, service.node_uuid
+    ):
+        return await monitor.get_stack_status(str(service.node_uuid))
+
+    # Service naming schema:
+    # NOTE: name is max 63 characters
+    # dy-sidecar_4dde07ea-73be-4c44-845a-89479d1556cf
+    # dy-revproxy_4dde07ea-73be-4c44-845a-89479d1556cf
+
+    # dynamic sidecar structure
+    # 0. a network is created: dy-sidecar_4dde07ea-73be-4c44-845a-89479d1556cf
+    # 1. a dynamic-sidecar is started: dy-sidecar_4dde07ea-73be-4c44-845a-89479d1556cf
+    # a traefik instance: dy-proxy_4dde07ea-73be-4c44-845a-89479d1556cf
+    #
+
+    service_name_dynamic_sidecar = assemble_service_name(
+        DYNAMIC_SIDECAR_SERVICE_PREFIX, service.node_uuid
+    )
+    proxy_service_name = assemble_service_name(
+        DYNAMIC_PROXY_SERVICE_PREFIX, service.node_uuid
+    )
+
+    # unique name for the traefik constraints
+    io_simcore_zone = f"{DYNAMIC_SIDECAR_SERVICE_PREFIX}_{service.node_uuid}"
+
+    # based on the node_id and project_id
+    dynamic_sidecar_network_name = (
+        f"{DYNAMIC_SIDECAR_SERVICE_PREFIX}_{service.node_uuid}"
+    )
+
+    # services where successfully started and they can be monitored
+    monitor_data = MonitorData.assemble(
+        service_name=service_name_dynamic_sidecar,
+        node_uuid=str(service.node_uuid),
+        project_id=service.project_id,
+        user_id=service.user_id,
+        hostname=service_name_dynamic_sidecar,
+        port=dynamic_services_settings.dynamic_sidecar.port,
+        service_key=service.key,
+        service_tag=service.version,
+        paths_mapping=simcore_service.paths_mapping,
+        compose_spec=simcore_service.compose_spec,
+        container_http_entry=simcore_service.container_http_entry,
+        dynamic_sidecar_network_name=dynamic_sidecar_network_name,
+        simcore_traefik_zone=io_simcore_zone,
+        service_port=TEMPORARY_PORT_NUMBER,
+        request_dns=x_dynamic_sidecar_request_dns,
+        request_scheme=x_dynamic_sidecar_request_scheme,
+        proxy_service_name=proxy_service_name,
+    )
```

Comment:
it's a bit of a huge command isn't it?
we discussed this, could we not have a pydantic model that simplify all this?

Reply:
Also the interface of monitoring could use NodeID like everywhere else in the director-v2? let's not mix str and int and uuid more than necessary.

Reply:
This was a long one! I've addressed all the requests!